docs/arrayloader-benchmarks.md

```diff
@@ -0,0 +1,183 @@
+
+## Non-sharded loading from local array backends
+
+To understand the performance difference in the large-scale benchmark, we thought of the simplest benchmark we could perform: measuring the time it takes to load a batch of data from a locally stored array into memory. Here, we use a single scRNA-seq expression matrix with shape 142k × 5k.
+
+Such a single dataset is typically generated in a single wetlab study that performs a small number of experiments. In the benchmark, accessing the array through HDF5 takes the least time, and zarr, tiledbsoma & parquet are at least a factor 10 slower (**Figure 2a**).
+
+![](arrayloader-benchmarks/figure_2a.svg)
+
+**Figure 2a ([source](https://lamin.ai/laminlabs/arrayloader-benchmarks/transform/1GCKs8zLtkc82llI))**: Benchmarking the time for loading batches of size 128 from a 142k × 5k array across 4 epochs and a range of array backends and configurations. The benchmark was run on AWS SageMaker on a `ml.g4dn.2xlarge` instance.
+
+Depending on the format, the dataset needs 100MB to 2.5GB of space on disk (**Figure 2b**).
+
+![](arrayloader-benchmarks/figure_2b.svg)
+
+**Figure 2b ([source](https://lamin.ai/laminlabs/arrayloader-benchmarks/core/transform/1GCKs8zLtkc82llI))**: Storage characteristics for the same dataset and array backends as in Figure 2a.
+
+The access pattern for all backends in this benchmark differs from Figure 1. In Figure 2a, simple dataloaders pull a single batch of random indices for hdf5, zarr, and tiledbsoma every iteration without collation or pre-loading. This differs from how `Merlin`, `MappedCollection`, and the `cellxgene-census` data loaders access the underlying data to generate batches. `MappedCollection` pulls single indices and collates them into batches and `cellxgene-census` preloads contiguous chunks of indices and then shuffles them to provide batches. `Merlin` also loads contiguous chunks of indices.
+
+## Scaling training across multiple GPUs
+
+We investigated whether `MappedCollection` would work well for scaling training across multiple GPUs. It works out of the box, and the speedup scales directly with the number of GPUs (**Figure 3**). Streamable datasets also scale across multiple GPUs, but typically require orchestrating workers.
+
+![](arrayloader-benchmarks/figure_3.svg)
+
+**Figure 3 ([source](https://lamin.ai/laminlabs/arrayloader-benchmarks/transform/AIJLqKqM0I4p5zKv))**: Samples per second loading from 9 `.h5ad` files with 775k samples in a Distributed Data-Parallel setup with 1, 2, and 4 GPUs.
```

Comment:
Maybe `Samples per second loading from 9 .h5ad files with 775k samples in a Distributed Data-Parallel setup with 1, 2, and 4 GPUs on NVIDIA A100 GPUs.`? In other examples the nodes used are provided, so it would make sense to specify GPU type as well.

Reply:
Will add it! I'm sorry I fear I removed this at some point in an iteration.

Reply:
https://github.com/laminlabs/lamin-blog/pull/17/commits/6ae904a63d7d331753673b8575c93da9385391a7